### ISSUE TITLE
model.detect gives same output for multiple different inputs

### ISSUE THREAD

**Issue by OpenCv30** [matterport/Mask_RCNN#1826]
Hi,
I am trying to run the model using 2 images at one time to increase performance.
Code is - <>/samples/demo.ipynb
`# Load a random image from the images folder
file_names = next(os.walk(IMAGE_DIR))[2]
image = skimage.io.imread(os.path.join(IMAGE_DIR, random.choice(file_names)))
image1 = skimage.io.imread(os.path.join(IMAGE_DIR, random.choice(file_names)))

# Run detection
inp=[]
print(image.shape)
inp.append(image)
inp.append(image1)
config.BATCH_SIZE=2
results = model.detect(inp,verbose=1) #[image], verbose=1)
# Visualize results
r = results[0]
visualize.display_instances(image, r['rois'], r['masks'], r['class_ids'], 
                            class_names, r['scores'])`
The outputs are coming same as show in the attached pdf at page 6 and 7.
[demo.pdf](https://github.com/matterport/Mask_RCNN/files/3774810/demo.pdf)

**Comment by barentine** [comment]
I am also running into this. @OpenCv30 did you ever get this sorted out?

**Comment by barentine** [comment]
Actually just learned you can't set `config.BATCH_SIZE` after instantiating the model, you need to have that set when you create the model, i.e.

```
config = Config()
config.BATCH_SIZE = 2
with tf.device(self.device):
    model = modellib.MaskRCNN(mode="inference", model_dir=self.model_dir, config=config)
```
should work, rather than
```
config = Config()
with tf.device(self.device):
    model = modellib.MaskRCNN(mode="inference", model_dir=self.model_dir, config=config)
model.config.BATCH_SIZE = 2
```

**Comment by OpenCv30** [comment]
Yes.. i have done same. I load model with default setting abd then decide batch-size= num of gpu * img per gpu. This worked. But, existing code doesn't handle the case when your images are less than batch size in last evaluation. One has to handle separately in model.detect()
